cordial_gui/scripts/gui_controller.py

```diff
@@ -148,44 +142,78 @@ def cb(self, msg):
                 if self.msg is None:
                     self.msg = msg
 
+        result = AskResult()
+
+        rospy.loginfo("Ask goal received: '" + goal.display.content + "'")
+
+        self._display_publisher.publish(goal.display)
+        self._gui_state = self.State.WAITING_FOR_USER_RESPONSE
+
+        self._last_active_time = datetime.datetime.now()
+
+        feedback_timer = rospy.timer.Timer(
+            rospy.Duration(
+                nsecs=int(
+                    1 * 1e9
+                )
+            ),
+            self._publish_action_feedback,
+        )
+
         wfm = WaitForMessage()
-        s = None
-        try:
-            s = rospy.topics.Subscriber(topic, topic_type, wfm.cb)
-            while (
-                    not rospy.core.is_shutdown()
-                    and wfm.msg is None
-                    and (datetime.datetime.now() - self._last_active_time).seconds < self._seconds_before_timeout
-                    and self._gui_state == self.State.WAITING_FOR_USER_RESPONSE
-            ):
-                rospy.rostime.wallsleep(0.01)
-        finally:
-            if s is not None:
-                s.unregister()
+        if not self._is_debug:
+            s = None
+            try:
+                s = rospy.topics.Subscriber(self._USER_RESPONSE_TOPIC, String, wfm.cb)
+                while(
+                        not rospy.core.is_shutdown()
+                        and wfm.msg is None
+                        and self._gui_state == self.State.WAITING_FOR_USER_RESPONSE
+                        and not self._prompt_action_server.is_preempt_requested()
+                ):
+
+                    rospy.rostime.wallsleep(0.01)
+
+            finally:
+                if s is not None:
+                    s.unregister()
+        else:
+            rospy.sleep(3)
```

Comment:
Why is this sleep here?  Can you use a variable for this so that it's more clear what it does?